src/stream_base-inl.h

```diff
@@ -77,8 +77,14 @@ void StreamBase::AddMethods(Environment* env,
 template <class Base>
 void StreamBase::GetFD(Local<String> key,
                        const PropertyCallbackInfo<Value>& args) {
-  StreamBase* wrap = Unwrap<Base>(args.Holder());
+  Base* handle;
 
+  // Mimic implementation of StreamBase::GetFD() and UDPWrap::GetFD().
+  ASSIGN_OR_RETURN_UNWRAP(&handle,
+                          args.Holder(),
+                          args.GetReturnValue().Set(UV_EINVAL));
+
+  StreamBase* wrap = static_cast<StreamBase*>(handle);
```

Comment:
It's hard to say whether this static_cast does the right thing.  `Base` can be a number of classes and each one has its own (frequently multiple) inheritance chain.

Not a fan of `StreamBase`, in case you're wondering.